### PR TITLE
Update DG and Check code quality

### PR DESCRIPTION
**Pull request by kaixiangtay**
Update DG, improve code quality in Command files and added java docs
in FindCommandParser

Changed primitive type Boolean to boolean type in DoneCommand

Refactor codes in if else block statements in AddCommand, 
DeleteFieldCommand, DeleteTaskCommand, EditCommand, PlanParser which
include boolean variables to improve code readability

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -259,12 +259,14 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 | Priority | As a …​ | I want to …​ | So that I can…​ |
 | -------- | ---------- | --------------- | ------------------ |
-| `* * *`  | new user | see usage instructions | refer to instructions when I forget how to use the App |
-| `* * *`  | user | add a new deadline | know when to complete it by |
+| `* * *`  | new user | see all possible commands | I know what I can do in the app |
```
Comment (maxxng) [inline docs/DeveloperGuide.md]: Not all commands will be inside. Unless we want to change the implementation

Comment (kaixiangtay) [reply docs/DeveloperGuide.md]: This is for list command, list all tasks the user has

Comment (maxxng) [reply docs/DeveloperGuide.md]: Refer to line 262

Comment (kaixiangtay) [reply docs/DeveloperGuide.md]: alright ok just for basic instructions then

**Review by maxxng — APPROVED** [review]
Looks good: code refactor and DG update

**Pull request merged**